addons/stock/static/src/stock_forecasted/stock_forecasted.js

```diff
@@ -0,0 +1,130 @@
+
+const { Component, onWillStart, useState, useSubEnv } = owl;
+
+class StockForecasted extends Component {
+    setup() {
+        useSubEnv({
+            ...this.env,
+            //ControlPanel trick : Allow the use of ControlPanel's bottom-right while disabling search to avoid errors
+            searchModel: {
+                searchMenuTypes: [],
+            },
+            config: {
+                ...this.env.config,
+                viewSwitcherEntries: [],
+            }
+        });
+
+        this.orm = useService("orm");
+        this.action = useService("action");
+
+        this.context = useState(this.props.action.context);
+        this.productId = this.context.active_id;
+        this.resModel = this.context.active_model;
+        this.title = this.props.action.name;
+
+        onWillStart(this._getReportValues);
+    }
+
+    async _getReportValues() {
+        await this._getResModel();
+        const isTemplate = !this.resModel || this.resModel === 'product.template';
+        this.reportModelName = `stock.forecasted_product_${isTemplate ? "template" : "product"}`;
+        const reportValues = await this.orm.call(this.reportModelName, "get_report_values", [], {
+            context: this.context,
+            docids: [this.productId],
+        });
+        this.docs = { ...reportValues.docs, ...reportValues.precision };
+    }
+
+    async _getResModel(){
+        this.resModel = this.context.active_model || (this.context.params && this.context.params.active_model);
+        //Following is used as a fallback when the forecast is not called by an action but through browser's history
+        if(!this.resModel){
+            if (this.props.action.res_model) {
+                const actionModel = await this.orm.read('ir.model', [Number(this.props.action.res_model)], ['model']);
+                if (actionModel.length && actionModel[0].model) {
+                    this.resModel = actionModel[0].model
+                }
+            }
+            else if (!this.resModel && this.props.action._originalAction) {
+                const originalContextAction = JSON.parse(this.props.action._originalAction).context;
+                if (originalContextAction) {
+                    this.resModel = originalContextAction.active_model || JSON.parse(originalContextAction.replace(/'/g, '"')).active_model;
+                }
+            }
+        }
+    }
+
+    async updateWarehouse(id) {
+        const hasPreviousValue = this.context.warehouse !== undefined;
+        this.context.warehouse = id;
+        if (hasPreviousValue) {
+            await this.reloadReport();
+        }
+    }
+
+    async reloadReport() {
+        return this.action.doAction(
+            {
+                type: "ir.actions.client",
+                tag: "stock_forecasted",
+                context: this.context,
+                name: this.title,
+            },
+            {
+                stackPosition: "replaceCurrentAction",
+            }
```

Comment:
```suggestion
        const actionRequest = {
            type: "ir.actions.client",
            tag: "stock_forecasted",
            context: this.context,
            name: this.title,
        };
        const options = { stackPosition: "replaceCurrentAction" };
        return this.action.doAction(actionRequest, options);
```
That way, it's a little shorter and it's easier to inspect the two objects before the call of `doAction`